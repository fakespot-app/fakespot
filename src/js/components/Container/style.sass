.container
  display: grid
  grid-template-columns: 300px 1fr 400px
  grid-template-areas: "beforeMain main afterMain"
