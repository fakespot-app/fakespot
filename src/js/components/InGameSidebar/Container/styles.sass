.container
  display: flex
  flex-direction: column
  width: 100%
  height: 100%
  background: $sidebar-background
  color: $sidebar-text-color
  