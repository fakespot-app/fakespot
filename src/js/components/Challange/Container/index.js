import React from "react";
import PropTypes from "prop-types";

import styles from "./styles.sass";

const Container = ({ children }) => (
  <main className={styles.container}>
    { children }
  </main>
);

Container.propTypes = {
  children: PropTypes.node,
};

Container.defaultProps = {
  children: null,
};

export default Container;