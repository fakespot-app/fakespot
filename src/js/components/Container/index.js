import React from "react";
import PropTypes from "prop-types";

import styles from "./style.sass";

const Container = ({ beforeMain, main, afterMain }) => (
  <div className={styles.container}>
    {beforeMain}

    <main className={styles.main}>
      {main}
    </main>

    {afterMain}
  </div>
);

Container.propTypes = {
  beforeMain: PropTypes.node,
  main: PropTypes.node,
  afterMain: PropTypes.node,
};

Container.defaultProps = {
  beforeMain: null,
  main: null,
  afterMain: null,
};

export default Container;
