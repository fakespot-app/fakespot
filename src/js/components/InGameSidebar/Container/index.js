import React from "react";
import PropTypes from "prop-types";

import styles from "./styles.sass";

const InGameSidebarContainer = ({ children }) => (
  <aside className={styles.container}>
    { children }
  </aside>
);

InGameSidebarContainer.propTypes = {
  children: PropTypes.node,
};

InGameSidebarContainer.defaultProps = {
  children: null,
};

export default InGameSidebarContainer;