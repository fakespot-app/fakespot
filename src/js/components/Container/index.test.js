import React from "react";
import { shallow, configure } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

import Container from "./index";

configure({ adapter: new Adapter() });

describe("<Container />", () => {
  test("should render", () => {
    const wrapper = shallow(<Container />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("should render beforeMain", () => {
    const beforeMain = (
      <span>
        Hello
      </span>
    );

    const wrapper = shallow(
      <Container
        beforeMain={beforeMain}
      />,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("should render main", () => {
    const main = (
      <span>
        There
      </span>
    );

    const wrapper = shallow(
      <Container
        main={main}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("should render afterMain", () => {
    const afterMain = (
      <span>
        Guys!
      </span>
    );

    const wrapper = shallow(
      <Container
        afterMain={afterMain}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("should render children and main prop", () => {
    const beforeMain = (
      <span>
        Hello
      </span>
    );

    const main = (
      <span>
        There
      </span>
    );

    const afterMain = (
      <span>
        Guys!
      </span>
    );

    const wrapper = shallow(
      <Container
        beforeMain={beforeMain}
        main={main}
        afterMain={afterMain}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});