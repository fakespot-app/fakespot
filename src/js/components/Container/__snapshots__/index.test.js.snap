// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should render 1`] = `
<div
  className="container"
/>
`;

exports[`<Container /> should render afterMain 1`] = `
<div
  className="container"
>
  <span>
    Guys!
  </span>
</div>
`;

exports[`<Container /> should render beforeMain 1`] = `
<div
  className="container"
>
  <span>
    Hello
  </span>
</div>
`;

exports[`<Container /> should render children and main prop 1`] = `
<div
  className="container"
>
  <span>
    Hello
  </span>
  <span>
    There
  </span>
  <span>
    Guys!
  </span>
</div>
`;

exports[`<Container /> should render main 1`] = `
<div
  className="container"
>
  <span>
    There
  </span>
</div>
`;
